### ISSUE TITLE
Spanish numbers

### ISSUE THREAD

**Issue by RParedesPalacios** [jpuigcerver/Laia#14]
since the run.sh failed i followed the instructions in README 

When running: 
../../laia-train-ctc \
  --adversarial_weight 0.5 \
  --batch_size "$batch_size" \
  --log_also_to_stderr info \
  --log_level info \
  --log_file laia.log \
  --progress_table_output laia.dat \
  --use_distortions true \
  --early_stop_epochs 100 \
  --learning_rate 0.0005 \
  model.t7 data/lang/chars/symbs.txt \
  data/train.lst data/lang/chars/train.txt \
  data/test.lst data/lang/chars/test.txt;

There are several incorrect paths because "*/chars/* should be "*/char/*"

**Comment by RParedesPalacios** [comment]
Also "$batch_size" is not defined

**Comment by RParedesPalacios** [comment]
And more, some "nan" and CER% larger than 100%:

[2017-01-14 18:13:07  INFO] ../../laia-train-ctc:283: Epoch 14, train summary: duration =     6s ; loss = 12601567730176036246469585403904.000000 ; cer = 590.12% ; del =   0.01% ; ins = 526.57% ; sub =  63.54% ; cer_ci = [564.99%,613.73%] ; ci_alpha =  5.000%
[2017-01-14 18:13:08  INFO] ../../laia-train-ctc:292: Epoch 14, valid summary: duration =     1s ; loss =      nan ; cer =  95.22% ; del =  13.45% ; ins =   7.23% ; sub =  74.54% ; cer_ci = [ 93.38%, 96.84%] ; ci_alpha =  5.000%
[2017-01-14 18:13:14  INFO] ../../laia-train-ctc:283: Epoch 15, train summary: duration =     6s ; loss =      nan ; cer = 587.06% ; del =   0.00% ; ins = 523.32% ; sub =  63.74% ; cer_ci = [561.74%,610.60%] ; ci_alpha =  5.000%
[2017-01-14 18:13:14  INFO] ../../laia-train-ctc:292: Epoch 15, valid summary: duration =     0s ; loss =      nan ; cer =  94.68% ; del =  14.40% ; ins =   6.47% ; sub =  73.81% ; cer_ci = [ 93.09%, 96.17%] ; ci_alpha =  5.000%

**Comment by jpuigcerver** [comment]
@RParedesPalacios which commit of Laia are you using? The last commit in master fixes these errors, AFAIK.

**Comment by RParedesPalacios** [comment]
I just clone it

git clone https://github.com/jpuigcerver/Laia.git

**Comment by jpuigcerver** [comment]
That is weird, according to https://github.com/jpuigcerver/Laia/blob/master/egs/spanish-numbers/run.sh batch_size is defined and the paths are correct.

Anyhow, I am more worried about the NaNs... In principle, CER% can be >100% because of insertion errors. Anyhow, I have never seen errors that big. Also, this kind of behavior tends to happen with very large learning rates, which should not be the case here (assuming that you are using batch_size=16 and learning_rate=0.0005).

Can you check that the input images look OK (maybe the preprocessing scripts are not working as expected), and try to disable the distortions (maybe the imgdistort library is not installed properly).

**Comment by RParedesPalacios** [comment]
I am not running run.sh i am copying the instructions that appear in README.MD

let me check all the params....

**Comment by RParedesPalacios** [comment]
Now, running run.sh, first epoch:
[2017-01-14 18:58:33  INFO] ../../laia-train-ctc:283: Epoch 1, train summary: duration =     6s ; loss =      nan ; cer = 590.25% ; del =   0.00% ; ins = 526.55% ; sub =  63.69% ; cer_ci = [565.17%,613.18%] ; ci_alpha =  5.000%
[2017-01-14 18:58:33  INFO] ../../laia-train-ctc:292: Epoch 1, valid summary: duration =     0s ; loss =      nan ; cer =  97.11% ; del =  92.85% ; ins =   0.00% ; sub =   4.26% ; cer_ci = [ 96.78%, 97.45%] ; ci_alpha =  5.000%
[2017-01-14 18:58:33  INFO] ../../laia-train-ctc:301: Epoch 1, new better model according to criterion "valid_cer": 0.971053 vs. inf (on epoch 0).

**Comment by RParedesPalacios** [comment]
and effectively model.t7 must be deleted to re-train

**Comment by dmartinalbo** [comment]
@RParedesPalacios, are you using a GTX1070? If so see https://github.com/baidu-research/warp-ctc/issues/40

**Comment by RParedesPalacios** [comment]
I am using GTX1080, the problem is that i installed warp-cdc with luarocks and i don't know how (and where) to modify the CMakeList.txt that they mentioned in that link….

**Comment by RParedesPalacios** [comment]
I have done… now it is working...

**Comment by RParedesPalacios** [comment]
Well,… more problems:

[2017-01-16 18:02:20  INFO] ../../laia-train-ctc:301: Epoch 16, new better model according to criterion "valid_cer": 0.759847 vs. 0.761025 (on epoch 15).
./run.sh: línea 46: 28847 Terminado (killed)      th ../../laia-train-ctc --adversarial_weight 0.5 --batch_size "$batch_size" --log_also_to_stderr info --log_level info --log_file laia.log --progress_table_output laia.dat --use_distortions true --early_stop_epochs 100 --learning_rate 0.0005 model.t7 data/lang/char/symbs.txt data/train.lst data/lang/char/train.txt data/test.lst data/lang/char/test.txt

process "Killed"

**Comment by jpuigcerver** [comment]
It looks like your process killed the process, but hard to know why without access to the logs... Possible reasons that I can think of immediately:
- Too much GPU memory.
- Too much CPU memory or CPU time and have some kind of limit.

**Comment by jpuigcerver** [comment]
Hi @RParedesPalacios ,

Apparently, some package that Laia uses (probably Torch or LuaJIT) has introduced a memory leak that happens each time the model is written to disk. I found the same problem as you after upgrading my Torch installation. @mauvilsa has an older version of Torch and he doesn't have this problem.

I'll try to find where the problem is and which Lua package is causing it.

@dmartinalbo If you recently installed Torch, you may be having this problem too.

**Comment by jpuigcerver** [comment]
Just update torch and you should be fine.
